Added new "Nerf Potions" effect

BMStrom opened the pull request:
Added new "Nerf Potions" effect that nerfs the player's current potions

BMStrom committed:
Added "Nerf Potions" effect

Added "Nerf Potions" effect, which looks at the materials in the player's potions and changes them into less useful materials

BMStrom committed:
Updated manifest for "Nerf Potions"

Updated manifest.json with files related to new "Nerf Potions" effect

Miczu committed:
Update manifest.json

Fix  Error downloading files/materials/spiritusigneus_materials.xml that should be
 Error downloading files/materials/spiritus_igneus_materials.xml
fixed tabulation while at it

Miczu commented:
@BMStrom looks like Soler has his own "flasks" in "data/entities/items/pickup/twitch_flask.xml" and those don't work with this.
Can you look if you can fix that? (maybe add the potion tag?).

Miczu commented:
Potential ideas:

blood -> slime
blood_worm -> urine (similar color and would be funny if he drinks it)
magic_liquid_random_polymorph -> magic_liquid_polymorph
magic_liquid_unstable_polymorph -> magic_liquid_polymorph
magic_liquid_invisibility -> water
midas -> gold_molten
mimic_liquid -> creepy_liquid (liquid that clones other, now liquid that clones itself)
magic_liquid_mana_regeneration -> acid (both liquids that eat metal, but rare good? upgrade)
magic_liquid_faster_levitation_and_movement -> magic_liquid_faster_levitation
magic_liquid_movement_faster -> magic_liquid_faster_levitation
magic_liquid_hp_regeneration -> magic_liquid_weakness (color and big funny)
magic_liquid_hp_regeneration_unstable ->  magic_liquid_weakness (color and big funny)
magic_liquid_invisibility ->blood_cold
milk -> molut (meme)
beer -> molut (meme)

Optional "more work" ideas:
magic_liquid_berserk -> glass cannon liquid??

BMStrom committed:
Fixed issues with twitch_flask and added more nerfs

Fixed issues with twitch_flask and added new materials to the ones affected by "Nerf Potions"

BMStrom committed:
Merge branch 'nerf-potions' of https://github.com/BMStrom/Noita-Twitch-Integration into nerf-potions

BMStrom committed:
Updated manifest with fixes

Update manifest.json with fixes and new potion nerfs

Miczu commented:
Flasks with unsupported liquids becomes empty? I think it's only fair to leave them as is or find a funny downgrade.
![noita-20250625-185926-358318357-00011501](https://github.com/user-attachments/assets/0fea9531-0c5e-4116-a6ed-83a1f622133f)

BMStrom committed:
Fixed unsupported potions being deleted

Fixed potions not in the "nerf" table being deleted from the player's flasks

BMStrom committed:
Updated manifest with fix

Updated manifest with fix for potion content deletion

Miczu commented:
![image](https://github.com/user-attachments/assets/a380490e-4100-42d9-8df6-6351def88985)
Noticed this message spam in noita_dev when I nerfed ambrosia

Miczu commented:
Looking at glue curse, it should be a dash and not an underscore
```
shoot_projectile( entity_who_caused, "mods/twitch-integration/data/entities/projectiles/weak_glue_shot.xml", pos_x, pos_y, 0, 0 )
```

Miczu commented on files/entities/misc/effect_weak_ambrosia.xml:
```diff
@@ -0,0 +1,32 @@
+	</LuaComponent>
+	
+	<LuaComponent
+		script_source_file="mods/spiritusigneus_stuff/files/scripts/status_effects/weak_ambrosia_end.lua"
```
mods/spiritusigneus_stuff doesn't look like twitch integration file path

Miczu commented on files/spiritusigneus_status_list_append.lua:
```diff
@@ -0,0 +1,35 @@
+		ui_name="Weak Ambrosia",
+		ui_description="Reduces all damage by half",
+		ui_icon="mods/twitch-integration/files/ui_gfx/status_indicators/protection_half.png",
+		effect_entity="mods/twitch_integration/files/entities/misc/effect_weak_ambrosia.xml"
```
wrong character spotted

Miczu commented on files/entities/misc/effect_weak_ambrosia.xml:
```diff
@@ -0,0 +1,32 @@
+    </InheritTransformComponent>	
+    
+    <LuaComponent
+		script_source_file="mods/spiritusigneus_stuff/files/scripts/status_effects/weak_ambrosia_start.lua"
```
Same here

BMStrom replied:
I don't know how I missed those, but they should be fixed now.

BMStrom committed:
Fixed filepaths for weak ambrosia effect

Fixed filepaths for weak ambrosia effect. Incorrect filepaths were left over from switching from my test mod to the twitch-integration mod

BMStrom committed:
Updated manifest with weak ambrosia fix

Updated manifest with weak ambrosia fix
